hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockNodeConnectionManager.java

```diff
@@ -532,15 +594,135 @@ public boolean selectNewBlockNodeForStreaming(final boolean force) {
     private @Nullable BlockNodeConfiguration findAvailableNode(@NonNull final List<BlockNodeConfiguration> nodes) {
```

Comment:
There's a lot happening inside this method, and it's private too. Given the number of possible configurations the CN could be in when this is called, it seems really hard to test comprehensively...

Reply:
I wondered if it's worth extracting out into a separate class, but as long as there's good coverage it's fine